.azure-pipelines/azure-pipelines-osx.yml

```diff
@@ -5,7 +5,7 @@
 jobs:
 - job: osx
   pool:
-    vmImage: macOS-12
+    vmImage: macOS-13
```

Comment:
@xylar The problem is here. The `vmImage` shouldn't be `macOS-13`.

The `MACOSX_SDK_VERSION` and the `c_stdlib_version` won't make a difference for as long as it's running on macOS 13. The bug is in the OS, not the SDK. Is there any way to change this (since this is an auto-generated file)?

We need to either change this to macOS-14, or disable the test.

Reply:
After failing to find a way to set the `vmImage`, I asked chatgpt, which claims that we can't control this parameter, as it is set globally for the conda-forge build system. Is this true? While chatgpt does make up a lot of nonsense, I suspect it might be right.

I suggest just skipping the test.

Reply:
Oh, I see!  Yes, I'm not aware of a way to change that.

I'm away until Tuesday but happy to add a skip for that test when I get back.